scio-examples/src/main/scala/com/spotify/scio/examples/complete/game/GameStats.scala

```diff
@@ -27,82 +27,117 @@ import org.apache.beam.sdk.options.StreamingOptions
       .withSideInputs(spammyUsers)
-      .filter { case (i, s) => s(spammyUsers).contains(i.user) }
+      // Filter out spammy users from this list -- `s(spammyUsers)` accesses the side input
+      .filter { case (i, s) => !s(spammyUsers).contains(i.user) }
+      // Done using the side input, convert back to regular `SCollection`
       .toSCollection
+      // Change each event into a tuple of: team user was on, and that user's score
       .map(i => (i.team, i.score))
+      // Sum the scores across the defined window, using "team" as the key to sum by
       .sumByKey
+      // Convert to `WindowedSCollection` to get windowing information with values
       .toWindowed
       .map { wv =>
+        // Convert windowed score to a `TeamScoreSums` object with windowing info as well as
+        // team and score info
         val start = fmt.print(wv.window.asInstanceOf[IntervalWindow].start())
         val now = fmt.print(Instant.now())
         wv.copy(value = TeamScoreSums(wv.value._1, wv.value._2, start, now))
       }
+      // Done using windowing information, convert back to regular `SCollection`
       .toSCollection
+      // Save to the BigQuery table defined by "output" in the arguments passed in + "_team" suffix
       .saveAsTypedBigQuery(args("output") + "_team")
 
     userEvents
+      // Window over a variable length of time - sessions end after sessionGap minutes no activity
       .withSessionWindows(
         Duration.standardMinutes(sessionGap),
         options = WindowOptions(timestampCombiner = TimestampCombiner.END_OF_WINDOW))
+      // Get all distinct users
       .keys.distinct
       .withWindow[IntervalWindow]
+      // Get duration of all sessions in minutes, discard user info
       .map { case (_, w) =>
         new Duration(w.start(), w.end()).toPeriod().toStandardMinutes.getMinutes
       }
+      // Find the mean value for user session length durations in a fixed time window
       .withFixedWindows(Duration.standardMinutes(userActivityWindowDuration))
       .mean
       .withWindow[IntervalWindow]
       .map { case (mean, w) =>
+        // Convert data on session length to `AvgSessionLength` case class, for BigQuery storage
         AvgSessionLength(mean, fmt.print(w.start()))
       }
+      // Save to the BigQuery table defined by "output" + "_sessions" suffix
       .saveAsTypedBigQuery(args("output") + "_sessions")
 
+    // Close context and run the job
     val result = sc.close()
+    // Wait to finish processing before exiting when streaming pipeline is halted
     exampleUtils.waitToFinish(result.internal)
   }
   // scalastyle:on method.length
 
   def calculateSpammyUsers(userScores: SCollection[(String, Int)]): SCollection[(String, Int)] = {
+    // Sum of scores by user
     val sumScores = userScores.sumByKey
+    // Average of all user scores
     val globalMeanScore = sumScores.values.mean
     sumScores
+      // Replicates global mean score to all workers - "cross product"
```

Comment:
I know in doc of `cross` we say `workers` but let's not do it here pls. it could be:

```
Cross product of global mean and user scores, effectively appending global mean to each (user, score) tuple.
```

does that make sense?

Reply:
yes, thanks! will use that language